raidpackage.py

```diff
@@ -0,0 +1,155 @@
+    initPost.add_field(name="Delivery and payment", value="An officer will mail you the items, and message you the price to deposit into the guild bank before the raid begins.", inline=False)
+    initMsg = await order_channel.send(embed=initPost)
+    await initMsg.add_reaction(icon_definitions.REACTION_NEW_ORDER)
+    
+    def check_channel_and_user_not_client(payload):
+        return str(payload.emoji) == icon_definitions.REACTION_NEW_ORDER and payload.channel_id == order_channel.id and payload.user_id != client.user.id
+    
+    reaction_payload = await client.wait_for('raw_reaction_add', check=check_channel_and_user_not_client)
+    usr = await client.fetch_user(reaction_payload.user_id)
+    sent_message, preorder_embed = await create_dm_preorder(usr, reaction_payload, client)
+
+
+    #Clean up reaction
+    await initMsg.remove_reaction(icon_definitions.REACTION_NEW_ORDER, usr)
+
+    async def listen(client, sent_message, preorder_embed, usr, confirmed_channel):
+        #Process Order Reactions
+        pending_reactions = [client.wait_for('raw_reaction_remove', check=lambda payload: payload.user_id != client.user.id),
+                client.wait_for('raw_reaction_add', check=lambda payload: payload.user_id != client.user.id)]
+        processed_reactions, pending_reactions = await asyncio.wait(pending_reactions, return_when=asyncio.FIRST_COMPLETED)
+
+        for processed_reaction in processed_reactions:
+            payload = await processed_reaction
+            if payload.event_type == 'REACTION_ADD':
+                await wait_for_order_reaction_add(payload, client, sent_message, preorder_embed, usr, confirmed_channel)
+
+            if payload.event_type == 'REACTION_REMOVE':
+                await wait_for_order_reaction_remove(payload, sent_message, preorder_embed)
+            
+            if str(payload.emoji) == icon_definitions.REACTION_ACCEPT and payload.channel_id != confirmed_channel:
+                user = await client.fetch_user(payload.user_id)
+                preorder_embed.title = "Confirmed RaidPackage Order"
+                preorder_embed.description = "Chosen Consumable Package:"
+                order_posting = await confirmed_channel.send(embed=preorder_embed)
+                await order_posting.add_reaction('✅')
+                await order_posting.add_reaction('💵')
+                await user.send(f"Your order is confirmed: {order_posting.jump_url}")
+                await sent_message.delete()
+                return
+
+            if str(payload.emoji) == icon_definitions.REACTION_CANCEL:
+                await cancel_order(sent_message, payload.user_id)
+                return
+
+        await listen(client, sent_message, preorder_embed, usr, confirmed_channel)
```

Comment:
i am not certain about this recursion. i tried to leverage wait_for() functionality but it captures only the first event so we need to loop over. Maybe reverting to separating raidpackage functionality over multiple event calls would be better if we added a persistent datastore to persist user_id/order_id